gempython/tools/xdaq/setDAQParamsPostConfigure.py

```diff
@@ -3,24 +3,90 @@
 from gempython.tools.amc_user_functions_uhal import *
 from gempython.utils.gemlogger import printGreen, printRed, printYellow
 
+def convert2Int(inputStr):
+    if 'x' in inputStr:
+        base = 16
+    elif 'b' in inputStr:
+        base = 2
+    else:
+        base = 10
+
+    return int(inputStr,base)
+
 def setDAQParams(args):
     from gempython.utils.gracefulKiller import GracefulKiller
     killer = GracefulKiller()
 
+    # Get ENV variable
+    from gempython.utils.wrappers import envCheck
+    envCheck("GEM_ADDRESS_TABLE_PATH")
+    import os
+    gemAddrPath = os.getenv("GEM_ADDRESS_TABLE_PATH")
+
+    # Determine INPUT_ENABLE_MASK
+    dict_ohMask = {}
+    import sys
+    if os.path.isfile("{}/gem-shelf{:02d}-input-en-mask.txt".format(gemAddrPath,args.shelf)):
+        printGreen("Loading existing input enable mask file at: $GEM_ADDRESS_TABLE_PATH/gem-shelf{:02d}-input-en-mask.txt".format(args.shelf))
+        # Load masks
+        inptEnMaskFile = open("{}/gem-shelf{:02d}-input-en-mask.txt".format(gemAddrPath,args.shelf),"r")
+        for row,line in enumerate(inptEnMaskFile):
+            # Skip commented lines
+            if line[0] == "#":
+                continue
+
+            # Parse line
+            line = line.strip("\n")
+            slotList = line.split(",")
+
+            # Get slot
+            try:
+                thisSlot = convert2Int(slotList[0])
+            except ValueError as err:
+                printRed("Reading $GEM_ADDRESS_TABLE_PATH/gem-shelf{:02d}-input-en-mask.txt: Invalid value given for AMC slot {}".format(args.shelf,slotList[0]))
+                inptEnMaskFile.close()
+                sys.exit(os.EX_DATAERR)
+
+            # Get ohMask
+            try:
+                dict_ohMask[thisSlot] = convert2Int(slotList[1])
```

Comment:
Likewise here, what prevents an `ohMask` of `0xcafe` of being parsed and applied here?
I guess you probably only use the "valid" bits later, but you could reasonably emit a warning here to at least alert on a parsed value that someone *thought* was OK

Reply:
thrown in a check that if this is greater than `0xfff` it's reset to `0xfff`.